### PR TITLE
Update docs related to build changes

### PR DESCRIPTION
**Pull request by mppf**
This PR includes docs updates related to PRs #19565 and #20396. This PR documents:
 * changes to how quickstart works
 * CHPL_LLVM_SUPPORT and the llvm support library being required.

While there, I:
 * adjusted prereqs.rst to only list the supported LLVM versions once and to say it is "LLVM and clang" that is needed
 * added sections in chplenv.rst for CHPL_LLVM_SUPPORT, CHPL_LLVM_CONFIG, and CHPL_LLVM_GCC_PREFIX

Reviewed by @lydia-duncan - thanks!

### CODE REVIEW

File: doc/rst/usingchapel/QUICKSTART.rst [hunk 1]
```diff
@@ -89,20 +89,22 @@ rebuild Chapel from source in a different configuration:
 * Open up a new shell to avoid inheriting the previous environment
   settings.
 
-* The Quickstart configuration described above sets ``CHPL_LLVM=none``
-  for simplicity and to save time.  This causes the Chapel compiler to
-  use its C back-end, which is not the preferred option; as of Chapel
-  1.25, LLVM is the default back-end, which needs to be available for
+* The Quickstart configuration attempts to detect if you have a
+  compatable system installation of LLVM and clang. If you do not, it
+  will ``CHPL_LLVM=none`` for simplicity and to save time.  This causes
```
Comment (lydia-duncan) [inline doc/rst/usingchapel/QUICKSTART.rst:94]: "will ``CHPL_LLVM=none``" -> "will set ``CHPL_LLVM=none``"

File: doc/rst/usingchapel/chplenv.rst [hunk 2]
```diff
@@ -767,32 +769,73 @@ CHPL_LLVM
...
      * ``unset`` otherwise
 
    If CHPL_LLVM is ``unset`` you will need to either add a system-wide
    installation of LLVM or set ``CHPL_LLVM`` to ``bundled`` or ``none``.
 
    See :ref:`readme-prereqs` for more information about currently
-   supported LLVM versions.
+   supported LLVM versions and how to install them. If you are having
+   trouble getting the build system to recognize your system install of
+   LLVM, try setting ``CHPL_LLVM=system`` and ``CHPL_LLVM_CONFIG`` as
```
Comment (lydia-duncan) [inline doc/rst/usingchapel/chplenv.rst:782]: What should `CHPL_LLVM_CONFIG` be set to?  Maybe linking to the new section will be sufficient

File: doc/rst/usingchapel/chplenv.rst [hunk 3]
```diff
@@ -754,9 +754,11 @@ CHPL_LLVM
        Value          Description
        ============== ======================================================
        bundled        use the llvm/clang distribution in third-party
-       system         find a compatible LLVM in system libraries;
-                      note: the LLVM must be a version supported by Chapel
+       system         find a compatible LLVM and clang in system libraries;
+                      but note that it must be a version supported by Chapel
        none           do not support llvm/clang-related features
+                      (but note that the LLVM Support library will still
+                      be used -- see ``CHPL_LLVM_SUPPORT`` below)
```
Comment (lydia-duncan) [inline doc/rst/usingchapel/chplenv.rst:761]: I think this could be a link to the section below, but it's okay to leave it as is.

**Review by lydia-duncan — APPROVED** [review]
Just a couple of improvements, otherwise I think this is good to go!

File: doc/rst/usingchapel/chplenv.rst (hunk 3 above)
Comment (mppf) [reply doc/rst/usingchapel/chplenv.rst:761]: Yeah I was worried the link in the table would be visually distracting for somebody reading the .rst file directly. I'll leave this one as is.

File: doc/rst/usingchapel/chplenv.rst (hunk 2 above)
Comment (mppf) [reply doc/rst/usingchapel/chplenv.rst:782]: I adjusted this section a bit.

File: doc/rst/usingchapel/chplenv.rst [hunk 4]
```diff
@@ -779,9 +779,10 @@ CHPL_LLVM
    See :ref:`readme-prereqs` for more information about currently
    supported LLVM versions and how to install them. If you are having
    trouble getting the build system to recognize your system install of
-   LLVM, try setting ``CHPL_LLVM=system`` and ``CHPL_LLVM_CONFIG`` as
-   well in order to get an error message that may help you to diagnose
-   the problem.
+   LLVM, try setting ``CHPL_LLVM=system`` and set ``CHPL_LLVM_CONFIG`` to
+   the `llvm-config` command from the LLVM version you have installed.
```
Comment (lydia-duncan) [inline doc/rst/usingchapel/chplenv.rst:783]: Should this be surrounded in two backticks on either side?

**Pull request merged**